checkpoint.go

```diff
@@ -193,27 +202,53 @@ func Check(p *CheckParams) (*CheckResponse, error) {
 // CheckInterval is used to check for a response on a given interval duration.
 // The interval is not exact, and checks are randomized to prevent a thundering
 // herd. However, it is expected that on average one check is performed per
-// interval. The returned channel may be closed to stop background checks.
-func CheckInterval(p *CheckParams, interval time.Duration, cb func(*CheckResponse, error)) chan struct{} {
-	doneCh := make(chan struct{})
+// interval.
+func CheckInterval(p *CheckParams, interval time.Duration,
+	cb func(*CheckResponse, error)) *CheckIntervalState {
+
+	state := &CheckIntervalState{
+		doneCh: make(chan struct{}),
+	}
 
-	if disabled := os.Getenv("CHECKPOINT_DISABLE"); disabled != "" {
-		return doneCh
+	if IsCheckDisabled() {
+		return state
```

Comment:
Sure, I was expecting the client code to handle that TBH - ie being able to differentiate between checkpoint enabled or not (and show it appropriately).  But I don't feel super strongly about it.